### DISCUSSION TITLE
PorousFlow: Sink dirackernel of components with the whole mass_flux

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#24375] [Q&A Modules: General] [answered]
Dear all, 

I learn the [Point and line sources/sinks](https://mooseframework.inl.gov/modules/porous_flow/sinks.html) page. I can set the source dirackernels with known separate mass_flux for all components by [PorousFlowSquarePulsePointSource](https://mooseframework.inl.gov/source/dirackernels/PorousFlowSquarePulsePointSource.html). For example,
```
[DiracKernels]
  [source1]
    type = PorousFlowSquarePulsePointSource
    start_time = 0
    end_time = 100
    point = '0.5 0.5 0'
    mass_flux = 0.1
    variable = pp
  []
  [source0]
    type = PorousFlowSquarePulsePointSource
    start_time = 0
    end_time = 100
    point = '0.5 0.5 0'
    mass_flux = 0.02
    variable = mf_0
  []
[]
```
And I also want to set the sink dirackernel for all components. Only the sum of the mass flux of all components is known. I try the [PorousFlowPeacemanBorehole](https://mooseframework.inl.gov/source/dirackernels/PorousFlowPeacemanBorehole.html) but the produced mass flux is dynamic and it is the function of the pressure. 

But I want to set the fixed overall mass flux of all components, equal to the overall mass flux in the source dirackernel. 

How can I solve this problem?

Thanks for your help.

J

**Comment by cpgr** [comment, marked as answer]
You could use a `PorousFlowPolyLineSink` and specify `mass_fraction_component` in place of the `PorousFlowPeacemanBorehole` as your production well - see https://github.com/idaholab/moose/blob/next/modules/combined/examples/geochem-porous_flow/geotes_2D/porous_flow.i for an example

**Reply by Joseph-0123** [reply]
> You could use a `PorousFlowPolyLineSink` and specify `mass_fraction_component` in place of the `PorousFlowPeacemanBorehole` as your production well - see https://github.com/idaholab/moose/blob/next/modules/combined/examples/geochem-porous_flow/geotes_2D/porous_flow.i for an example

Hello Chris, thanks for your suggestions. It does work well.
